### PR TITLE
Unavailability calendar in pdf version of application records

### LINKED ISSUE #264: Add unavailability calendar to print view
The PDF print view doesn't include the applicant's unavailability. Time for moar Prawn!

### PR DESCRIPTION
**Pull request by bgregg**
^^that. Closes #264.

### CODE REVIEW

**Comment by bgregg** [comment]
screen shot 2016-08-22 at 11 59 36 am

File: spec/factories/unavailabilities.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 FactoryGirl.define do
   factory :unavailability do
```
Comment (dfaulken) [inline spec/factories/unavailabilities.rb]: Does the factory need to be this complex?

File: app/models/unavailability.rb [hunk 2]
```diff
@@ -6,4 +6,13 @@ class Unavailability < ActiveRecord::Base
   # this will serialize the columns of the unavailability model
   # so that they accept strings in the form of arrays
   Date::DAYNAMES.map(&:downcase).map(&:to_sym).each { |d| serialize d, Array }
+
+  def grid
```
Comment (dfaulken) [inline app/models/unavailability.rb]: I know I wrote this, but a comment explaining what type of data structure this method returns could be useful.

File: spec/factories/unavailabilities.rb (hunk 1 above)
Comment (bgregg) [inline spec/factories/unavailabilities.rb]: Probably not. I did this as opposed to putting it directly in the spec

File: spec/models/application_record_spec.rb [hunk 3]
```diff
@@ -19,6 +19,21 @@
     end
   end
 
+  describe 'unavailability_rows' do
+    before :each do
+      @record = create :application_record, :with_unavailability
+    end
+    let :call do
```
Comment (dfaulken) [inline spec/models/application_record_spec.rb]: This could also be a one-liner.

Comment (dfaulken) [inline spec/models/application_record_spec.rb]: As in

```ruby
let(:call){ @record.unavailability.grid }
```

File: spec/models/application_record_spec.rb [hunk 4]
```diff
@@ -19,6 +19,21 @@
     end
   end
 
+  describe 'unavailability_rows' do
+    before :each do
+      @record = create :application_record, :with_unavailability
```
Comment (dfaulken) [inline spec/models/application_record_spec.rb]: Prefer `let` blocks to instance variables.

File: spec/factories/unavailabilities.rb (hunk 1 above)
Comment (dfaulken) [inline spec/factories/unavailabilities.rb]: Better to put it in the spec, for two reasons:

1. You don't need to read two different files to determine why a test is passing or failing;
2. A factory should only have the absolute minimum information required to create a valid instance of an object (that is, it should echo the model validations).

File: app/pdfs/print_record_pdf.rb [hunk 5]
```diff
@@ -5,15 +5,15 @@ class PrintRecordPdf < Prawn::Document
   def initialize(record)
     super()
     @record = record
```
Comment (bgregg) [inline app/pdfs/print_record_pdf.rb]: I guess I don't really know what you mean. I see what you mean by removing that line, but then how will we access the record in methods other than initialize()?

Comment (dfaulken) [inline app/pdfs/print_record_pdf.rb]: In the same way that `unavailability_calendar` does, by passing it as a method argument

File: spec/factories/unavailabilities.rb (hunk 1 above)
Comment (dfaulken) [inline spec/factories/unavailabilities.rb]: (I see you moved the logic to the spec, but it's still in this file too 👅 )

**Pull request merged**